pkg/server/cloud/aws/instances.go

```diff
@@ -441,6 +466,41 @@ func (e *AwsEC2) retrieveOrAllocateHost(node *api.Node) (*string, error) {
 	return allocateOutput.HostIds[0], nil
 }
 
+func (e *AwsEC2) isHostAvailable(hostId *string) bool {
+	describeOutput, err := e.client.DescribeHosts(&ec2.DescribeHostsInput{
+		HostIds: aws.StringSlice([]string{aws.StringValue(hostId)}),
+	})
+	if err != nil {
+		klog.Errorf("cannot describe host %s: %v", *hostId, err)
+		return false
+	}
+	if len(describeOutput.Hosts) != 1 {
+		klog.Errorf("no such host: %s", aws.StringValue(hostId))
+		return false
+	}
+	host := describeOutput.Hosts[0]
+	return aws.StringValue(host.State) == AwsInstanceAvailableState
+}
+
+func (e *AwsEC2) waitForHostAvailable(ctx context.Context, hostId *string) bool {
+	hostAvailable := false
+	ticker := time.NewTicker(1 * time.Second)
+	for {
+		select {
+		case <-ctx.Done():
+			return hostAvailable
+		case <-ticker.C:
+			hostAvailable = e.isHostAvailable(hostId)
+			if hostAvailable {
+				return hostAvailable
+			}
+			klog.V(2).Infof("host %s not available yet", *hostId)
+		default:
```

Comment:
Don't leave in the `default` case, otherwise it will always be taken while the other two cases are not true, effectively busy looping here.

Reply:
:+1: removed